sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/regexpExpressions.scala

```diff
@@ -68,7 +68,20 @@ trait StringRegexExpression extends ImplicitCastInputTypes {
  * Simple RegEx pattern matching function
  */
 @ExpressionDescription(
-  usage = "str _FUNC_ pattern - Returns true if str matches pattern and false otherwise.")
+  usage = "str _FUNC_ pattern - Returns true if str matches pattern, " +
+    "null if any arguments are null, false otherwise.",
+  extended =
+    "The pattern is a string which is matched literally, with exception to the " +
```

Comment:
Triple quotes would help in the source, however if want correct indentation I would need to use pipes and call `.stripMargin` on the string. That is unfortunately not possible since the string would then not be considered a constant anymore (constants are required for annotation parameters)